plugins/ban.py

```diff
@@ -67,13 +76,20 @@ def unban(self, msg, args):
         room_data = json.loads(r.text)
         unbanned_rooms = []
 
-        for room in filter(lambda x: x.get('uri', None) in joined_rooms,
-                           room_data):
-            if room is not None:
+        if room_data == []:
+            yield 'No rooms found:('
+        else:
+            for room in filter(lambda x: x.get('uri', None) in joined_rooms,
+                               room_data):
                 url = 'https://api.gitter.im/v1/rooms/' + \
                     room['id'] + '/bans/' + sinner
                 rq = requests.delete(url, headers=headers)
                 if rq.status_code == 200:
                     unbanned_rooms.append(room['uri'])
+                else:
+                    self.log.info('Error ' + str(rq.status_code))
+                    yield 'Error ' + str(rq.status_code) + \
+                          ': Something went wrong:( Pls try again!'
```

Comment:
Please add changes as mentioned above, not repeating them here to keep it simple.